### ISSUE TITLE
$page.url is undefined 

### ISSUE THREAD

**Issue by dreitzner** [sveltejs/kit#8047]
### Describe the bug

When building production with cloudflare adapter there seems to be an issue with `$page.url` being undefined during SSR.
This happened after deploying to cloudflare, dev environment was fine.

My component structure:
```
+layout.svelte
  Nav.svelte
```
The code that throws the error
```
<!-- Will give error; TypeError: Cannot read properties of undefined (reading 'pathname')-->
<a
	rel="prefetch"
	aria-current={'/music' === $page.url.pathname ? 'page' : undefined}
	href="/music"
>
	music
</a>
```

### Reproduction

https://github.com/dreitzner/reitzner.at/tree/37facc1953261e7a5a56d0ff7d1ced60072798a8 (repo before commit 5701c86232b8e048194bd42194e7bb297f90382d)

1. npm run build
2. check if wrangler is installed (if not run `npm install --global wrangler`)
3. run `wrangler pages dev .\.svelte-kit\cloudflare\` in project root
4. press `b` to open browser
5. see internal server error

### Logs

```Shell
From wrangler:

TypeError: Cannot read properties of undefined (reading 'pathname')
    at C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\entries\pages\_layout.svelte.js:16:139
    at Object.$$render (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\chunks\index.js:86:18)
    at C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\entries\pages\_layout.svelte.js:60:34
    at Object.$$render (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\chunks\index.js:86:18)
    at C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\index.js:39:97
    at $$render (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\chunks\index.js:86:18)
    at Object.render (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\chunks\index.js:94:20)
    at render_response (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\index.js:1123:29)
    at render_page (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\index.js:1597:12)
    at resolve (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\index.js:2327:22)
TypeError: Cannot read properties of undefined (reading 'pathname')
    at C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\entries\pages\_layout.svelte.js:16:139
    at Object.$$render (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\chunks\index.js:86:18)
    at C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\entries\pages\_layout.svelte.js:60:34
    at Object.$$render (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\chunks\index.js:86:18)
    at C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\index.js:39:97
    at $$render (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\chunks\index.js:86:18)
    at Object.render (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\chunks\index.js:94:20)
    at render_response (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\index.js:1123:29)
    at respond_with_error (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\index.js:1382:12)
    at render_page (C:\Users\projects\private\reitzner.at\.svelte-kit\output\server\index.js:1613:12)
GET / 500 Internal Server Error (123.95ms)
```

### System Info

```Shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 AMD Ryzen 7 PRO 5850U with Radeon Graphics
    Memory: 15.68 GB / 31.56 GB
  Binaries:
    Node: 16.18.0 - ~\AppData\Local\Volta\tools\image\node\16.18.0\node.EXE
    Yarn: 1.22.18 - ~\AppData\Local\Volta\tools\image\yarn\1.22.18\bin\yarn.CMD
    npm: 8.19.2 - ~\AppData\Local\Volta\tools\image\node\16.18.0\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (108.0.1462.42)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.90
    @sveltejs/adapter-cloudflare: ^1.0.0-next.41 => 1.0.0-next.41
    @sveltejs/kit: next => 1.0.0-next.580
    svelte: ^3.44.0 => 3.54.0
```

### Severity

serious, but I can work around it

**Comment by sanqi** [comment]
$page.params is also the same error

**Comment by gtm-nayan** [comment]
@dreitzner @sanqi can you check again with the latest version of SvelteKit?

**Comment by sanqi** [comment]
@gtm-nayan  now is working fine.

**Comment by dummdidumm** [comment]
Probably another symptom of #7950 which was fixed through #8092
